### PR TITLE
Reload tree when imodel changes

### PR DESCRIPTION
**Pull request by saskliutas**
Closes https://github.com/iTwin/presentation/issues/57

Prerequisites: 
- https://github.com/iTwin/imodel-native/pull/208
- https://github.com/iTwin/itwinjs-core/pull/5219

Changed Tree update logic after imodel is changed. Now whole Tree is reloaded and use same reloader that is used to reload Tree after ruleset or ruleset variables changes.

### CODE REVIEW

File: packages/components/src/presentation-components/tree/controlled/TreeHooks.ts [hunk 1]
```diff
@@ -197,33 +182,18 @@ function useModelSourceUpdateOnIModelHierarchyUpdate(params: {
...
+          subscription = reloadTree(modelSource.getModel(), dataProvider, pageSize, renderedItems.current).subscribe({
+            next: (newModelSource) => setTreeNodeLoaderState((prevState) => ({
               modelSource: newModelSource,
               rulesetRegistration: prevState.rulesetRegistration,
-              dataProvider: newDataProvider,
-            }));
-          }
+              dataProvider,
+            })),
+          });
```
Comment (grigasp) [inline packages/components/src/presentation-components/tree/controlled/TreeHooks.ts:196]: looks like this part could be reused by all 3 update-related hooks

**Review by grigasp — APPROVED** [review]

File: packages/components/src/presentation-components/tree/controlled/TreeReloader.ts [hunk 2]
```diff
@@ -59,54 +65,90 @@ class TreeReloader extends PagedTreeNodeLoader<IPresentationTreeDataProvider> {
...
+  }
+
+  private reloadPreviouslyExpandedNodes(previouslyExpandedNodes: ExpandedNode[]) {
+    return from(previouslyExpandedNodes)
+      .pipe(
+        // Process expanded nodes recursively, breadth first
+        expand((expandedNode) => {
+          const node = this.modelSource.getModel().getNode(expandedNode.id);
+          if (node !== undefined) {
+          // The expanded node is already loaded in the new tree model, now load and expand its children recursively
```
[the remark this answers is not in the document]
Comment (saskliutas) [reply packages/components/src/presentation-components/tree/controlled/TreeReloader.ts]: Fixed

File: packages/components/src/presentation-components/tree/controlled/TreeHooks.ts (hunk 1 above)
Comment (saskliutas) [reply packages/components/src/presentation-components/tree/controlled/TreeHooks.ts]: Extracted same code to separate function

**Pull request merged**